utils/driverless.c

```diff
@@ -626,13 +540,85 @@ list_printers (int mode, int reg_type_no, int isFax)
     if (debug)
       fprintf(stderr, "DEBUG: Started %s (PID %d)\n", ippfind_argv[0],
 	      ippfind_ipp_pid);
-    listPrintersInArray(post_proc_pipe_ipp, service_uri_list_ipps, 0, mode,
-			isFax);
   }
+  /*
+  Reading the ippfind output in CUPS Array
+  */
+  if(reg_type_no >=1)
+  {
+    dup2(post_proc_pipe_ipps[0], 0);
+    close(post_proc_pipe_ipps[0]);
+    close(post_proc_pipe_ipps[1]);
 
-  close(post_proc_pipe_ipp[0]);
-  close(post_proc_pipe_ipp[1]);
+    fp = cupsFileStdin();
 
+    while ((bytes = cupsFileGetLine(fp, buffer, sizeof(buffer))) > 0) 
+    {
+      ippfind_output = (char *)malloc(8192*(sizeof(char)));
+      ptr = buffer;
+
+      while (ptr && !isalnum(*ptr & 255)) ptr ++;
+
+      if ((!strncasecmp(ptr, "ipps", 4) && ptr[4] == '\t'))
+      {
+        ptr += 4;
+        *ptr = '\0';
+        ptr ++;
+      } else
+      continue;
+      snprintf(ippfind_output,8191,"%s",ptr);
```

Comment:
No need to use '8191' - ```snprintf``` automatically uses ```n-1``` for string and on the index ```n``` is a NULL terminator - see f.e. [cplusplus page](http://www.cplusplus.com/reference/cstdio/snprintf/). You can use a name for 8192.